pkg/apis/elasticsearch/v1/elasticsearch_types.go

```diff
@@ -172,6 +172,9 @@ type TransportTLSOptions struct {
 	// - `ca.crt`: The CA certificate in PEM format.
 	// - `ca.key`: The private key for the CA certificate in PEM format.
 	Certificate commonv1.SecretRef `json:"certificate,omitempty"`
+	// CertificateAuthorities is a references to a config map that contains one or more x509 certificates for
+	// trusted authorities in PEM format. The certificates need to be in a file called `ca.crt`.
```

Comment:
I wonder if there are cases where the name of the key is not under the control of the user. For example the CA key name from the [OpenShift CA service](https://docs.openshift.com/container-platform/4.12/security/certificate_types_descriptions/service-ca-certificates.html#purpose) is `service-ca.crt`:

> Other services can request that the CA bundle for the service CA be injected into API service or config map resources by annotating with service.beta.openshift.io/inject-cabundle: true to support validating certificates generated from the service CA. In response, the Operator writes its current CA bundle to the CABundle field of an API service or **as service-ca.crt to a config map**

I don't think however that the OpenShift CA service can be used to manage the transport certificates, so it's not a valid argument. And we can offer a first implementation where the expected key is `ca.crt`, and later add a new field to select a different key:

```yaml
  transport:
    tls:
      certificateAuthorities:
        configMapName: trust
        keys: ## Optional, ca.crt by default
          - service-ca.crt
```